### ISSUE TITLE
Error: A 'declare' modifier cannot be used in an already ambient context.

### ISSUE THREAD

**Issue by jvilk** [SitePen/dts-generator#14]
First off: Great tool! It seems like it is very close to working for my use case.

I'm using TypeScript 1.4.1, and it looks like `dts-generator` incorrectly nests `declare` statements. For example, here's the internal module generated from the declaration file for the external module 'gLong' in one of my projects:

```{typescript}
declare module 'doppio/src/gLong' {
        declare class gLong {
          // ...
        }
        export = gLong;
}
```

It looks like `dts-generator` needs to remove inner `declare` modifiers.

**Comment by jvilk** [comment]
Doing `:%s/\tdeclare/\t/g` and `:%s/\texport declare/\texport/g` in vim removed those issues, but now I'm having different trouble:

```
error TS2307: Cannot find external module './jvm'.
error TS2439: Import declaration in an ambient external module declaration cannot reference external module through relative external module name.
```

Seems like dts-generator doesn't rewrite external module imports with relative module names to the new canonical names.

**Comment by mtraynham** [comment]
Also seeing this per upgrade to 1.3

**Comment by csnover** [comment]
Hm, I’ll take a look, problems like this usually happen when someone changes the SyntaxKind enum in TypeScript and you are using a different version of TypeScript, so the code to skip declare keywords is no longer followed.

**Comment by mtraynham** [comment]
I'm pulling `master` down through npm, so I'll try with 1.5.0-alpha.  Most of my "modules" are just classes, so once I removed the inner `declare` it worked as noted by @jvilk.  Although, I didn't see the other error.

```
declare module 'foo/User' {
    declare class User {
        name: string;
        roles: Array<string>;
        constructor(data?: Object);
        toJSON(): Object;
    }
    export = User;
}
```
to:
```
declare module 'foo/User' {
    class User {
        name: string;
        roles: Array<string>;
        constructor(data?: Object);
        toJSON(): Object;
    }
    export = User;
}
```

**Comment by csnover** [comment]
Yeah, SyntaxKind changed again and I forgot to update the typescript.d.ts before the last build. This is fixed in 1.3.1 (and fixed so it can’t break again). Sorry about that!
